packages/taco-auth/src/providers/eip712.ts

```diff
@@ -64,6 +67,10 @@ export class EIP712AuthProvider implements AuthProvider {
     private readonly provider: ethers.providers.Provider,
     private readonly signer: ethers.Signer,
   ) {
+    console.warn(
+      'DeprecationWarning: The default authentication provider, EIP712AuthProvider, is deprecated. ' +
+      'Please use EIP4361AuthProvider instead. Refer to the documentation for more details.'
+    );
```

Comment:
Since the use EIP712 was implicit until now, is this necessary?

`EIP712AuthProvider` was not actually used/specified by anyone in prior releases. Can we simply remove it?

Reply:
IMHO it doesn't matter if behavior is implicit or explicit, only whether the changes in the API are breaking, and this is a breaking change

Reply:
I'm not sure I follow. I believe this file/module and the class was only added as part of the unreleased epic (the epic we are merging this PR into), so why keep the file/module and the class that we no longer intend to use as part of that epic and was never released...

By "implicit" I mean that no user has ever specified or created a EIP712AuthProvider object, because it was never released. So there is no real "breaking" for this class/module since has never been released.  The broader API (`decrypt`) will indeed have "breaking" changes but that is irrespective of this particular auth provider, or the scheme used for validating wallets. 

It's possible that a warning could be used somewhere, but I don't think this is the spot, since it seems like this file/module should just be deleted.

Reply:
We've released an API with a documented behavior (using EIP712 by default) and are now replacing that behavior with another (EIP4361). I don't know specifically how this change can affect our users (is it going to "break" anything) but changing the implicit (or explicit) behavior of an API is a breaking change (according to semver).

Deprecating EIP712 in T+1 major version will allow us to communicate this breaking change to users who are currently using EIP712. Again, I don't know whether it will actually break something, but it makes sense to announce it and only remove it in T+2 after this upcoming change is communicated.

Reply:
I see. Ok, so the goal of keeping the file is more to keep this (EIP712)  as an option (albeit a pending deprecating one), in case some adopter decides they want to use EIP712 instead of EIP4361 for `:userAddress`. That's the reason for keeping this module for now.

If we are affording that option, then the server-side will need to be adjusted to accept both EIP712 and EIP4361 for `:userAddress` and https://github.com/nucypher/nucypher/pull/3515 will need to be adjusted.

Reply:
So the server side verification will be broken for EIP712 in the next release? Not deprecated, but outright removed?

In that case, these changes have no sense and we should remove EIP712 from the client too. I will address it in this PR.

Reply:
> So the server side verification will be broken for EIP712 in the next release? Not deprecated, but outright removed?

That's still a discussion for #3515, and not currently set in stone. Still need to make an official decision there.